### ISSUE TITLE
Expected lua string when using menu border

### ISSUE THREAD

**Issue by vuki656** [MunifTanjim/nui.nvim#85]
Hey,

some recent commit seems to have broken the border settings in the menu component.

I was using it like this (note the border setup):

```lua
local menu = Menu({
    relative = "cursor",
    border = {
        style = "rounded",
        highlight = "Normal",
        text = {
            top = options.title,
            top_align = "left",
        },
    },
    position = {
        row = 0,
        col = 0,
    },
    size = {
        width = 50,
        height = 2,
    },
    highlight = "Normal:Normal",
    focusable = true,
}, {
    lines = {
        Menu.item(PROMPT_ACTIONS.confirm),
        Menu.item(PROMPT_ACTIONS.cancel),
    },
    keymap = {
        focus_next = { "j", "<Down>", "<Tab>" },
        focus_prev = { "k", "<Up>", "<S-Tab>" },
        close = { "<Esc>", "<C-c>" },
        submit = { "<CR>", "<Space>" },
    },
    on_submit = function(selected_action)
        if selected_action.text == PROMPT_ACTIONS.confirm then
            utils.job({
                json = false,
                command = options.command,
                on_success = function()
                    options.on_submit()
                end,
                on_error = function()
                    options.on_cancel()
                end,
            })
        else
            options.on_cancel()
        end
    end,
})
```

but now when the component is mounted it throws an error:
![image](https://user-images.githubusercontent.com/8104698/147748307-0b0c41c5-b955-4b7f-b374-ee39230ba971.png)

--- 

I even tried using the default example border but still got the same error.

```lua
local menu = Menu({
    relative = "cursor",
    border = {
        style = "rounded",
        text = {
            top = "[Choose Item]",
            top_align = "center",
        },
    },
    position = {
        row = 0,
        col = 0,
    },
    size = {
        width = 50,
        height = 2,
    },
    highlight = "Normal:Normal",
    focusable = true,
}, {
    lines = {
        Menu.item(PROMPT_ACTIONS.confirm),
        Menu.item(PROMPT_ACTIONS.cancel),
    },
    keymap = {
        focus_next = { "j", "<Down>", "<Tab>" },
        focus_prev = { "k", "<Up>", "<S-Tab>" },
        close = { "<Esc>", "<C-c>" },
        submit = { "<CR>", "<Space>" },
    },
    on_submit = function(selected_action)
        if selected_action.text == PROMPT_ACTIONS.confirm then
            utils.job({
                json = false,
                command = options.command,
                on_success = function()
                    options.on_submit()
                end,
                on_error = function()
                    options.on_cancel()
                end,
            })
        else
            options.on_cancel()
        end
    end,
})
```

---

Specifically, it seems that `text` inside the `border` is causing the problem since when I comment it out and just leave `style` it works as expected.

**Comment by MunifTanjim** [comment]
Hey @vuki656 , sorry about that... a tiny issue fell through the cracks. The fix is pushed at 76fc8edf771adff74aba513bd0f62e984996ef88

I'll add automated test for this later so that it doesn't happen again.

**Comment by vuki656** [comment]
> Hey @vuki656 , sorry about that... a tiny issue fell through the cracks. The fix is pushed at [76fc8ed](https://github.com/MunifTanjim/nui.nvim/commit/76fc8edf771adff74aba513bd0f62e984996ef88)
> 
> I'll add automated test for this later so that it doesn't happen again.

That was quick! Thanks for the fix.
